pymodm/fields.py

```diff
@@ -1062,8 +1062,27 @@ def to_python(self, value):
             return self.related_model.from_document(value)
         return value
 
+    @staticmethod
+    def _value_to_mongo(related_model, value):
+        if isinstance(value, bson.SON):
+            return value
+
+        if isinstance(value, related_model):
+            return value.to_son()
+
+        # if value is a dict convert in to model
+        # so we can properly generate SON
+        if isinstance(value, dict):
+            return related_model.from_document(value).to_son()
```

Comment:
I see your point. `to_son` [iterates the fields](https://github.com/mongodb/pymodm/blob/ce603f5079fb602eeb2421e1bca767f4cb2bd8ac/pymodm/base/models.py#L281-L288) of the model class and outputs keys in a deterministic order, as you noted. That makes sense now. Carry on!